docs/UPGRADE-4.0.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: page
+title: Upgrading from 3.x to 4.0
+---
+
+This document outlines the steps one should take when upgrading to the new major version.
+
+## Step 1: Review the new requirements
+
+Requirements have changed, this project now requires **PHP 7.4 or later** and **Moodle 3.8.3** or later.
+
+## Step 2: Review the change log
+
+Detailed information about what changed in Version 4 can be found in the [change log](CHANGELOG.md).
+
+## Step 3: Review the CIs configuration files
+
+### GitHub Actions
+
+Review the updated [gha.dist.yml](https://github.com/moodlehq/moodle-plugin-ci/blob/master/gha.dist.yml)
+and update the `.github/workflows/ci.yml` (or whatever it's named) file in your plugin. For detailed information about the contents of that `ci.yml` file, please see this [help document](GHAFileExplained.md).
+
+A summary of the actions required, really simple for version 4, is:
+
+1. Change the `composer create-project` line and bump to from version 3 (^3) to version 4 (^4).
+2. Test it and, hopefully, party!
+
+### Travis CI
+
+Review the updated [.travis.dist.yml](https://github.com/moodlehq/moodle-plugin-ci/blob/master/.travis.dist.yml)
+and update your `.travis.yml` file in your plugin. For detailed information about the contents of `.travis.dist.yml`
+file, please see this [help document](TravisFileExplained.md).
+
+1. Change the `composer create-project` line and bump to from version 3 (^3) to version 4 (^4).
+2. Test it and, hopefully, party!
+
+## FAQ
+
+### What is happening to Version 3?
+
+Version 3 still exists within its own [3.x](https://github.com/moodlehq/moodle-plugin-ci/tree/3.x) branch. You can continue using it, but it is no longer getting new features and may not receive additional updates.  In addition, it may start breaking in Moodle 4.3 or later, because of 8.2 incompatibilities.
+
+### Why Version 4?
+
+Mainly to be able to modernise all the internal components of the application that, basically were the same since version 2, making it really future-ready. And to leave behind very old versions of PHP (< 7.4) and Moodle (< 3.8.3), not supported since long ago.
```

Comment:
```suggestion
Mainly to be able to modernise all the internal components of the application which had remained largely unmodified since version 2. This will make it much easier to plan for the future and to leave behind very old versions of PHP (< 7.4) and Moodle (< 3.8.3), which have not been supported for a long time.
```